### ISSUE TITLE
cross compile w/ mingw

### ISSUE THREAD

**Issue by braydonf** [json-c/json-c#295]
With tagged version `json-c-0.12.1-20160607`:
```
./autogen.sh
./configure --host=x86_64-w64-mingw32 --enable-shared
make
```
There is an error and failure to compile:
```
random_seed.c:184:0: error: ignoring #pragma comment  [-Werror=unknown-pragmas]
 #pragma comment(lib, "advapi32.lib")
```

Is json-c intended to cross compile using x86_64-w64-mingw32, and etc.?

**Comment by braydonf** [comment]
This is the next error after clearing that first one:
```
.libs/arraylist.o: In function `array_list_expand_internal':
/home/bgf/vendor/json-c/arraylist.c:67: undefined reference to `rpl_realloc'
.libs/json_object.o: In function `json_object_new_string_len':
/home/bgf/vendor/json-c/json_object.c:724: undefined reference to `rpl_malloc'
.libs/printbuf.o: In function `printbuf_new':
/home/bgf/vendor/json-c/printbuf.c:42: undefined reference to `rpl_malloc'
.libs/printbuf.o: In function `printbuf_extend':
/home/bgf/vendor/json-c/printbuf.c:72: undefined reference to `rpl_realloc'
/home/bgf/vendor/json-c/printbuf.c:72: undefined reference to `rpl_realloc'
collect2: error: ld returned 1 exit status
```

**Comment by braydonf** [comment]
Found a solution that clears the rpl_realloc undefs:
```
./autogen.sh
ac_cv_func_malloc_0_nonnull=yes ./configure --host=x86_64-w64-mingw32 --enable-shared
make
```

**Comment by ploxiln** [comment]
the `rpl_malloc` problem was fixed in #216 (which was not backported to the 0.12 branch)

**Comment by ploxiln** [comment]
actually that wasn't it, sorry, just having trouble finding it now, it was a ways back ...

**Comment by ploxiln** [comment]
not sure if it was a PR but this was the fix 10d50aadf2a0803aea95fa92632d229b872f2cf3

**Comment by braydonf** [comment]
I had tried cross compile on the master branch also, but ran into another issue:

```
arraylist.c:30:2: error: #error Unable to determine size of size_t
 #error Unable to determine size of size_t
  ^
arraylist.c: In function ‘array_list_expand_internal’:
arraylist.c:76:20: error: ‘SIZE_T_MAX’ undeclared (first use in this function)
   if( arr->size >= SIZE_T_MAX / 2 )
                    ^
arraylist.c:76:20: note: each undeclared identifier is reported only once for each function it appears in
arraylist.c: In function ‘array_list_put_idx’:
arraylist.c:95:13: error: ‘SIZE_T_MAX’ undeclared (first use in this function)
   if (idx > SIZE_T_MAX - 1 ) return -1;
```

**Comment by hawicz** [comment]
So how big *is* size_t on your platform?  Can you compile and run a small program that does something like?:
    printf("%d, s:%d i:%d l:%d ll:%d\n", sizeof(size_t), sizeof(short), sizeof(int), sizeof(long), sizeof(long long));

**Comment by braydonf** [comment]
Yeah, here are the results:

```
x86_64-w64-mingw32-gcc -o test.exe test.c
wine test.exe
> 8, s:2 i:4 l:4 ll:8
```

```
i686-w64-mingw32-gcc -o test.exe test.c
wine test.exe
> 4, s:2 i:4 l:4 ll:8
```

**Comment by hawicz** [comment]
Please try it again with the most recent code.  That should help allow it to compile arraylist.c at least.  I wouldn't be surprised if other issues pop up with trying to run on a LLP64 platform, but maybe we'll get lucky and it'll just work.

**Comment by braydonf** [comment]
Yes that works, and compiles with:
```
./configure --host=x86_64-w64-mingw32
```

I tried running tests also with `make check`, and ran into some other issues. Made a few changes and  it worked (though not sure the correct changes):
https://gist.github.com/braydonf/41091d8a63d111aa2d954dccb1ad0d56

**Comment by braydonf** [comment]
Closing, as this issue is fixed on master. Will open another for tests.
